### PR TITLE
Destroy hanging xcodebuild process before starting a new one

### PR DESCRIPTION
**Pull request by rafael-chavez**

### CODE REVIEW

**Comment by rafael-chavez** [comment]
@imurchie

File: lib/webdriveragent.js [hunk 1]
```diff
@@ -95,6 +97,21 @@ class WebDriverAgent {
     this.proxyReqRes = this.jwproxy.proxyReqRes.bind(this.jwproxy);
   }
 
+  async destroyXcodeProcesses () {
+    await exec('ps', ['-aux']).then(
```
Comment (imurchie) [inline lib/webdriveragent.js]: This should be 

```js
try {
  let {stdout} = await exec('ps', ['-aux'];
  let match = ...
} catch (err) {
  log.error(`Could not get xcodebuild processes: ${err}`);
}
```

**Comment by imurchie** [comment]
Do the pre-commit hooks not work for you? The Travis job is failing for linting errors, which ought to prevent a commit as well.

File: WebDriverAgent [hunk 2]
```diff
@@ -1 +1 @@
-Subproject commit 6813e2549b684379e1b6a6ee96deb2b42f4e1a85
+Subproject commit 9bfb4c6bc59c71a3a522af02418c99dae4ada6dd
```
Comment (imurchie) [inline WebDriverAgent]: This commit is 2 months old. Not sure why it's here.

https://github.com/facebook/WebDriverAgent/commit/9bfb4c6bc59c71a3a522af02418c99dae4ada6dd

**Comment by rafael-chavez** [comment]
@imurchie travis is failing is this expected? 
Im updating the submodule pointer to a more recent commit than what you have here https://github.com/appium/appium-xcuitest-driver/pull/172

File: WebDriverAgent (hunk 2 above)
Comment (rafael-chavez) [inline WebDriverAgent]: fixed

File: lib/webdriveragent.js (hunk 1 above)
Comment (rafael-chavez) [inline lib/webdriveragent.js]: fixed

**Comment by imurchie** [comment]
I would recommend against updating WDA within a PR that is dealing with separate things. It is often not straightforward.

**Comment by imurchie** [comment]
Travis's Mac cloud has been weird of late. I'm working on some stuff to try to fix some issues for us, but it remains that things are flakey.

**Comment by rafael-chavez** [comment]
removed wda update @imurchie

**Comment by imurchie** [comment]
Awesome! Thanks for this!

**Pull request merged**